delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala

```diff
@@ -45,26 +45,38 @@ final class BlazegraphCoordinator private (
   def run(offset: Offset): Stream[Task, Elem[Unit]] = {
     fetchViews(offset).evalMap { elem =>
       elem
-        .traverse {
-          case active: ActiveViewDef =>
-            IndexingViewDef
-              .compile(
-                active,
-                compilePipeChain,
-                graphStream,
-                sink(active)
-              )
-              .flatMap { projection =>
-                cleanupCurrent(active.ref) >>
-                  supervisor.run(
-                    projection,
-                    for {
-                      _ <- createNamespace(active)
-                      _ <- cache.put(active.ref, active)
-                    } yield ()
+        .traverse { v =>
+          cache.get(v.ref).flatMap { cachedView =>
+            (cachedView, v) match {
+              case (Some(cached), active: ActiveViewDef) if cached.projection == active.projection =>
+                for {
+                  _ <- cache.put(active.ref, active)
+                  _ <- Task.delay {
+                         logger.info(s"Index ${active.projection} already exists and will not be recreated.")
+                       }
+                } yield ()
+              case (_, active: ActiveViewDef)                                                      =>
+                IndexingViewDef
+                  .compile(
+                    active,
+                    compilePipeChain,
+                    graphStream,
+                    sink(active)
                   )
-              }
-          case d: DeprecatedViewDef  => cleanupCurrent(d.ref)
+                  .flatMap { projection =>
+                    cleanupCurrent(active.ref) >>
+                      supervisor.run(
+                        projection,
+                        for {
+                          _ <- createNamespace(active)
+                          _ <- cache.put(active.ref, active)
+                        } yield ()
+                      )
+                  }
+              case (_, deprecated: DeprecatedViewDef)                                              =>
+                cleanupCurrent(deprecated.ref)
```

Comment:
I missed it in the ES part but you don't have to do another look up to get the value in the cache

Reply:
Corrected in both